rollup.config.js

```diff
@@ -1,37 +1,54 @@
-import commonjs from 'rollup-plugin-commonjs';
-import nodeResolve from 'rollup-plugin-node-resolve';
-import nodeBuiltins from 'rollup-plugin-node-builtins';
-import babel from 'rollup-plugin-babel';
-import banner from 'rollup-plugin-banner';
-import genHeader from './lib/header';
+import commonjs from 'rollup-plugin-commonjs'
+import resolve from 'rollup-plugin-node-resolve'
+import babel from 'rollup-plugin-babel'
+import typescript from 'rollup-plugin-typescript2'
+import banner from 'rollup-plugin-banner'
+import genHeader from './lib/header'
 
-export default [{
-	input: 'index.js',
-	output: {
-		file: 'dist/esm.js',
-		format: 'esm',
-	},
-	plugins: [
-		commonjs(),
-		nodeBuiltins(),
-		nodeResolve(),
-		babel(),
-		banner(genHeader('data'))
-	],
-},
-{
-	input: 'index.js',
-	output: {
-		file: 'dist/umd.js',
-		format: 'umd',
-		exports: 'named',
-		name: 'vis',
-	},
-	plugins: [
-		commonjs(),
-		nodeBuiltins(),
-		nodeResolve(),
-		babel(),
-		banner(genHeader('data'))
-	],
-}]
+// TypeScript because Babel transpiles modules in isolation, therefore no type reexports.
+// CommonJS because Babel is not 100 % ESM.
+
+const babelConfingBase = {
+  extensions: ['.mts', '.ts', '.tsx', '.mjs', '.js', '.jsx'],
+  runtimeHelpers: true,
+}
+const resolveConfig = {
+  extensions: [...babelConfingBase.extensions, '.json', '.node'],
+}
+const bannerConfig = genHeader('data')
+const typescriptConfig = {
+  tsconfig: 'tsconfig.code.json',
+}
+
+export default [
+  {
+    input: 'src/index.ts',
+    output: {
+      file: 'dist/esm.js',
+      format: 'esm',
+    },
+    plugins: [
+      resolve(resolveConfig),
+      typescript(typescriptConfig),
+      commonjs(),
+      babel(babelConfingBase),
+      banner(bannerConfig),
+    ],
+  },
+  {
+    input: 'src/index.ts',
+    output: {
+      file: 'dist/umd.js',
+      format: 'umd',
+      exports: 'named',
+      name: 'visData',
+    },
+    plugins: [
+      resolve(resolveConfig),
+      typescript(typescriptConfig),
+      commonjs(),
+      babel(babelConfingBase),
+      banner(bannerConfig),
+    ],
+  },
+]
```

Comment:
I have to say that I'm not so happy with the complete rewrite of this file. It is basically the same as before but with typescript support. 
@Thomaash  Did you generate this file with a cli tool or did you rewrite it just for taste? But I guess I could live with the changes..

Reply:
I added TypeScript support and reformatted (most of the optical changes come from this). It's just to keep the same code style throughout the package.